.github/workflows/slow.yaml

```diff
@@ -134,3 +134,52 @@ jobs:
         with:
           name: build-logs-macos
           path: btlayer-*.log
+
+  freebsd:
+    strategy:
+      matrix:
+        osversion:
+          - 14.1
+          - 13.3
+
+    runs-on: ubuntu-22.04
+    name: freebsd(${{ matrix.osversion }})
+
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v4
+
+      - name: Run test-builds
+        id: test-builds
+        uses: vmactions/freebsd-vm@v1
+        with:
+          usesh: true
+          release: ${{ matrix.osversion }}
+          prepare: |
+            pkg update
+            pkg install -y \
+              autoconf \
+              autoconf-archive \
+              automake \
+              bash \
+              cppunit \
+              gmake \
+              libltdl \
+              libtool \
+              m4 \
+              nettle \
+              pkgconf \
+              translate-toolkit
+
+          run: |
+            export MAKE=gmake
+            export CC=clang
+            export CXX=clang++
```

Comment:
Now that ccache optimization is removed, why do we need these customizations? Please either remove them (if they are not needed) or add a YAML comment to explain them (otherwise).

Reply:
Without these, in practice, we would still use clang, but it would be called 'gcc'. 
Would that be confusing to you or anyone else looking at it?

Reply:
> Without these, in practice, we would still use clang, but it would be called 'gcc'.

Understood. If compiler naming is the only effect of these customizations, then please remove these customizations.

> Would that be confusing to you or anyone else looking at it?

I am sure that will be confusing or at least puzzling to some folks (possibly including me). However, customizing this confuses or puzzles some folks as well (obviously including me)! We are selecting among two evils here.

This PR is not about improving FreeBSD (or autoconf or whatever) approach to naming things. With all other factors being equal, the closer we stay to FreeBSD customs/conventions when testing on FreeBSD, the better, regardless of how problematic those customs/conventions are.

Reply:
> the closer we stay to FreeBSD customs/conventions when testing on FreeBSD,

we have no idea about what FreeBSD customs/conventions are. I suspect that just like MacOS does, they install portability shims to make nonportable software content.

I don't agree we should follow problematic customs or conventions, especially when it comes to explicitly naming what compiler we target. But this is not a hill I'm going to die for

Reply:
> > the closer we stay to FreeBSD customs/conventions when testing on FreeBSD,
> 
> we have no idea about what FreeBSD customs/conventions are.

In this context, we do: Whatever happens by default (i.e. without our customizations) is likely to be customary/conventional.

> I suspect that just like MacOS does, they install portability shims to make nonportable software content.

You may be right, but it is their business. Folks using FreeBSD ought to be accustomed to whatever FreeBSD is doing.

> I don't agree we should follow problematic customs or conventions, especially when it comes to explicitly naming what compiler we target.

With all other factors being equal, I think we should approximate, to the extent it is feasible/practical, what a typical admin sees/does when compiling Squid on FreeBSD. Any unnecessary customizations are unwanted in this "Does Squid build on FreeBSD?" context. We are not trying to build on FreeBSD configured "our way". We are trying to build on FreeBSD configured "their way" (again, to the extent it is feasible/practical).

> But this is not a hill I'm going to die for

IMO, this design decision can and should be made based on merits rather than personal sacrifices. And since many other decisions are based on the same basic "what are we testing?" question, I hope that this discussion helps reduce the time spent on re-arguing them in the future. Every time we are about to add a customization to a build environment, we have to pause and ask ourselves: Is it justified? Do our actions really have to differ from what admins are going to do? And, if yes, where do we explain the necessity of this customization?

Reply:
> > > the closer we stay to FreeBSD customs/conventions when testing on FreeBSD,
> > 
> > 
> > we have no idea about what FreeBSD customs/conventions are.
> 
> In this context, we do: Whatever happens by default (i.e. without our customizations) is likely to be customary/conventional.

These aren't FreeBSD defaults, these are autotools' defaults. If we were to base our project on - say - cmake or bzl, we might be seeing something different. If FreeBSD has any default, it's `cc` and `c++`, but autoconf is opinionated

> > I suspect that just like MacOS does, they install portability shims to make nonportable software content.
> 
> You may be right, but it is their business. Folks using FreeBSD ought to be accustomed to whatever FreeBSD is doing.
> 
> > I don't agree we should follow problematic customs or conventions, especially when it comes to explicitly naming what compiler we target.
> 
> With all other factors being equal, I think we should approximate, to the extent it is feasible/practical, what a typical admin sees/does when compiling Squid on FreeBSD. Any unnecessary customizations are unwanted in this "Does Squid build on FreeBSD?" context. We are not trying to build on FreeBSD configured "our way". We are trying to build on FreeBSD configured "their way" (again, to the extent it is feasible/practical).

As shown above, we aren't. I can agree to use CC=cc and CXX=c++, if that helps

> > But this is not a hill I'm going to die for
> 
> IMO, this design decision can and should be made based on merits rather than personal sacrifices. And since many other decisions are based on the same basic "what are we testing?" question, I hope that this discussion helps reduce the time spent on re-arguing them in the future. Every time we are about to add a customization to a build environment, we have to pause and ask ourselves: Is it justified? Do our actions really have to differ from what admins are going to do? And, if yes, where do we explain the necessity of this customization?

This is one of the decisions where it's hard or even impossible to agree based on merits, because we are comparing outcomes that boil down to different axes. On one side there's "let's be explicit about what we are testing against", and on the other there is "let's add as little context as possible". They are mutually exclusive, and achieve different goals. These goals cannot be ranked objectively, only subjectively; we are finding it hard to disagree because I think one goal is more valuable, and you think another one is. Our processes grant more power to the reviewer than to the author of a PR, therefore my statement, which could be expanded to "I disagree with your assessment, but I rank the goal of landing this PR more than I rank the goal of not landing it because I fixate on what I consider a valuable but minor detail, therefore I'll follow your opinions although I disagree with them".

So, let's have it your way and move on :)